app/models/merchant.rb

```diff
@@ -2,4 +2,35 @@ class  Merchant < ApplicationRecord
   has_many :items
 
   validates :name, presence: true
```

Comment:
Hi @dionew1 @BrianZanti - Wondered if I could get some input from one of you on this BI method.

I've been struggling for a good chunk of the afternoon with how to get this to work via Active Record. The raw SQL is working correctly as best I can tell -- my tests are passing so that's a semi-positive indicator. Below that in the commented out section is but one version of the many AR approaches I attempted. The subquery from lines 22-25 seems to be working as intended, but I'm struggling to join it back to a top level query to return the data I need. 

I've attempted doing this via .joins & .from, passing the subquery as a variable, as an interpolated string, etc. (The next commit shows a different variation of what I've attempted.)

Questions I have at this point are:

1) I assume you're looking for more Active Record & less straight-SQL on a method like this, so it is actually worth my time to go down this road, right?
2) Is this approach to do everything at Merchant model & pull data via a subquery the best type of approach? Is there some other preferable way to aggregate grandchild data that I'm just blanking on?
3) Do you have any suggestions on resources for researching what I'm trying to do? Not finding much from Google efforts that's panning out.

Thanks! (And, yes, i realize one of you are likely lecturing about this specific topic to some extent on Weds.)

Reply:
Hi Andy, thanks for taking the time to write out this question and give me some good context. Yes, this is worth your time to figure out in AR without doing a subquery. t looks like what you are doing is calculating the revenue for each invoice, and then totaling those invoices for each merchant. What you need to do is combine the logic from the two separate queries into one. How can you skip totaling the revenue for each invoice and go straight to totaling the revenue for each merchant? You will probably need to think through how to join all of these tables together to make the right associations.